starky/src/cross_table_lookup.rs

```diff
@@ -1019,34 +1037,35 @@ pub(crate) fn verify_cross_table_lookups_circuit<
     debug_assert!(ctl_zs_openings.iter_mut().all(|iter| iter.next().is_none()));
 }
 
-#[cfg(test)]
-pub(crate) mod testutils {
-    use std::collections::HashMap;
+#[cfg(debug_assertions)]
+pub mod testutils {
+    #[cfg(not(feature = "std"))]
+    use alloc::{vec, vec::Vec};
 
+    use hashbrown::HashMap;
     use plonky2::field::polynomial::PolynomialValues;
     use plonky2::field::types::Field;
 
-    use crate::all_stark::Table;
-    use crate::cross_table_lookup::{CrossTableLookup, TableWithColumns};
+    use super::{CrossTableLookup, TableIdx, TableWithColumns};
 
-    type MultiSet<F> = HashMap<Vec<F>, Vec<(Table, usize)>>;
+    type MultiSet<F> = HashMap<Vec<F>, Vec<(TableIdx, usize)>>;
 
     /// Check that the provided traces and cross-table lookups are consistent.
-    pub(crate) fn check_ctls<F: Field>(
+    pub fn check_ctls<F: Field>(
         trace_poly_values: &[Vec<PolynomialValues<F>>],
         cross_table_lookups: &[CrossTableLookup<F>],
-        extra_memory_looking_values: &[Vec<F>],
+        extra_looking_values: &HashMap<TableIdx, Vec<Vec<F>>>,
     ) {
         for (i, ctl) in cross_table_lookups.iter().enumerate() {
-            check_ctl(trace_poly_values, ctl, i, extra_memory_looking_values);
+            check_ctl(trace_poly_values, ctl, i, extra_looking_values.get(&i));
         }
     }
 
     fn check_ctl<F: Field>(
         trace_poly_values: &[Vec<PolynomialValues<F>>],
         ctl: &CrossTableLookup<F>,
         ctl_index: usize,
-        extra_memory_looking_values: &[Vec<F>],
+        extra_looking_values: Option<&Vec<Vec<F>>>,
```

Comment:
`&Vec<…>` is a bit annoying here, especially given that we clone the values on line 1091 anyway. Can we do better? Explicit clone in the caller?

Reply:
Yeah I agree, I am not too bothered by performances because this is never to be run in an actual release mode anyway (without extra debug assertions). But I'll look at changing it